File: application-client/src/main/java/io/wwan13/auth/controller/AuthController.java
package io.wwan13.auth.controller;

import io.wwan13.auth.dto.LoginRequestDto;
import io.wwan13.auth.dto.LoginResponseDto;
import io.wwan13.auth.dto.SignUpRequestDto;
import io.wwan13.auth.dto.SignUpResponseDto;
import io.wwan13.auth.usecase.LoginUseCase;
import io.wwan13.auth.usecase.SignUpUseCase;
import io.wwan13.constant.HttpStatusCode;
import io.wwan13.auth.dto.CreateProfileImageResponseDto;
import io.wwan13.auth.dto.GetProfileImageExamplesResponseDto;
import io.wwan13.auth.usecase.CreateProfileImageUseCase;
import io.wwan13.auth.usecase.GetProfileImageExamplesUseCase;
import io.wwan13.response.SuccessResponse;
import lombok.RequiredArgsConstructor;
import org.springframework.web.bind.annotation.*;

@RestController
@RequiredArgsConstructor
@RequestMapping("/auth")
public class AuthController {

    private final SignUpUseCase signUpUseCase;
    private final LoginUseCase loginUseCase;
    private final GetProfileImageExamplesUseCase getProfileImageExamplesUseCase;
    private final CreateProfileImageUseCase createProfileImageUseCase;

    @PostMapping("/signup")
    public SuccessResponse<SignUpResponseDto> signUp(
            @RequestBody SignUpRequestDto signUpRequestDto) {
        SignUpResponseDto response = signUpUseCase.execute(signUpRequestDto);
        return SuccessResponse.of(HttpStatusCode.CREATED, response);
    }

    @PostMapping("/login")
    public SuccessResponse<LoginResponseDto> login(
            @RequestBody LoginRequestDto loginRequestDto) {
        LoginResponseDto response = loginUseCase.execute(loginRequestDto);
        return SuccessResponse.of(HttpStatusCode.OK, response);
    }

    @GetMapping("/{memberEmail}/profileImage")
    public SuccessResponse<GetProfileImageExamplesResponseDto> getExamples(
            @PathVariable String memberEmail) {
        GetProfileImageExamplesResponseDto response = getProfileImageExamplesUseCase.execute(memberEmail);
        return SuccessResponse.of(HttpStatusCode.OK, response);
    }

    @PostMapping("/{memberEmail}/profileImage")
    public SuccessResponse<CreateProfileImageResponseDto> createProfileImage(
            @PathVariable String memberEmail,
            @RequestBody String profileImageUrl) {
        CreateProfileImageResponseDto response = createProfileImageUseCase.execute(memberEmail, profileImageUrl);
        return SuccessResponse.of(HttpStatusCode.CREATED, response);
    }
}

File: grida-core/core-api/src/main/kotlin/org/grida/presentation/v1/AuthController.kt
package org.grida.presentation.v1

import org.grida.api.ApiResponse
import org.grida.application.auth.LoginRequest
import org.grida.application.auth.LoginResponse
import org.grida.application.auth.LoginUseCase
import org.springframework.web.bind.annotation.PostMapping
import org.springframework.web.bind.annotation.RequestBody
import org.springframework.web.bind.annotation.RequestMapping
import org.springframework.web.bind.annotation.RequestParam
import org.springframework.web.bind.annotation.RestController

@RestController
@RequestMapping("/api/v1/auth")
class AuthController(
    private val loginUseCase: LoginUseCase
) {

    @PostMapping
    fun provideAuthToken(
        @RequestParam("platform") platform: String,
        @RequestBody request: LoginRequest
    ): ApiResponse<LoginResponse> {
        val response = loginUseCase.execute(platform, request)
        return ApiResponse.success(response)
    }
}
